lambda/getHistoryTs.ts

```diff
@@ -0,0 +1,43 @@
+import type {
+	SimDetails,
+	SIMNotExistingError,
+	SIMNotFoundError,
+} from './getSimDetailsFromCache.js'
+import type { simInfoTS } from './onomondo/getAllUsedSimsOnomondo.js'
+
+export const getHistoryTs =
+	({
+		getSimDetailsFromCache,
+	}: {
+		getSimDetailsFromCache: (iccid: string) => Promise<
+			| {
+					error: SIMNotFoundError | SIMNotExistingError
+			  }
+			| {
+					sim: SimDetails & {
+						ts: Date
+					}
+					historyTs?: Date
+			  }
+		>
+	}) =>
+	async (
+		iccid: string,
+		dataUsage: Record<string, Record<string, simInfoTS>>,
+	): Promise<{ oldHistoryTs: Date; newHistoryTs: Date }> => {
+		const cacheInfo = await getSimDetailsFromCache(iccid)
```

Comment:
Since the result of `getSimDetailsFromCache` depends on a value that is available outside of this function (`iccid`) AND the call is not conditional, this function can be simplified by passing in the result of `getSimDetailsFromCache(iccid)`.

Reply:
Then this entire function can be expressed as this:

```typescript
const newHistoryTs =  Object.entries(dataUsage[iccid]).pop() ?? (await getSimDetailsFromCache(iccid))?.historyTs ?? new Date(Date.now() - 60 * 1000 * 60 * 24)
```

Reply:
I made a PR to show what I mean: https://github.com/bifravst/sim-details/pull/88